src/bootstrap/bootstrap.py

```diff
@@ -217,29 +238,37 @@ def download_stage0(self):
                        match="rust-mingw", verbose=self.verbose)
 
         if self.cargo().startswith(self.bin_root()) and \
-                (not os.path.exists(self.cargo()) or self.cargo_out_of_date()):
-            self.print_what_it_means_to_bootstrap()
+                (not os.path.exists(self.cargo()) or
+                 self.program_out_of_date(self.cargo_stamp())):
+            self.print_what_bootstrap_means()
             filename = "cargo-{}-{}.tar.gz".format(cargo_channel, self.build)
-            url = self._download_url + "/dist/" + self.stage0_date()
+            url = "{}/dist/{}".format(self._download_url, self.stage0_date())
             tarball = os.path.join(rustc_cache, filename)
             if not os.path.exists(tarball):
                 get("{}/{}".format(url, filename),
                     tarball, verbose=self.verbose)
             unpack(tarball, self.bin_root(),
                    match="cargo", verbose=self.verbose)
-            self.fix_executable(self.bin_root() + "/bin/cargo")
-            with open(self.cargo_stamp(), 'w') as f:
-                f.write(self.stage0_date())
+            self.fix_executable("{}/bin/cargo".format(self.bin_root()))
+            with open(self.cargo_stamp(), 'w') as cargo_stamp:
+                cargo_stamp.write(self.stage0_date())
 
-    def fix_executable(self, fname):
-        # If we're on NixOS we need to change the path to the dynamic loader
+    @staticmethod
+    def fix_executable(fname):
+        """This method tries to fix the given executable
 
+        Please note that if we're on NixOS we need to change the path to the
+        dynamic loader
+        """
         default_encoding = sys.getdefaultencoding()
         try:
             ostype = subprocess.check_output(
                 ['uname', '-s']).strip().decode(default_encoding)
-        except (subprocess.CalledProcessError, WindowsError):
+        except subprocess.CalledProcessError:
             return
+        except OSError as reason:
+            if getattr(reason, 'winerror', None) is not None:
+                return
```

Comment:
@alexcrichton I think is not required to re-raise here because if the following condition is `true`:

```python
if getattr(reason, 'winerror', None) is not None:
```

Then, we can agree that we don't need to modify the interpreter section to fix the dynamic linker in the given executable. This condition only applies on NixOS. wdyt?